src/contract.rs

```diff
@@ -3,19 +3,21 @@ use crate::error::{std_err_result, ContractError};
 use crate::msg::{ExecuteMsg, InitMsg, MigrateMsg, NameResponse, NameSearchResponse, QueryMsg};
 use crate::state::{config, config_read, meta, meta_read, NameMeta, State};
 use cosmwasm_std::{
-    coin, from_binary, to_binary, Api, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo,
-    Order, Response, StdError, StdResult, Uint128,
+    coin, entry_point, from_binary, to_binary, Api, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env,
+    MessageInfo, Order, Response, StdError, StdResult, Uint128,
 };
 use cosmwasm_storage::Bucket;
 use provwasm_std::{
     add_attribute, bind_name, Attribute, Attributes, NameBinding, ProvenanceMsg, ProvenanceQuerier,
+    ProvenanceQuery,
 };
 
 ///
 /// INSTANTIATION SECTION
 ///
+#[entry_point]
```

Comment:
One of the biggest changes is that each exposed route (instantiate, query, execute, etc) requires this `entry_point` annotation now.  Trying to add a contract that is missing this annotation will fail, and adding a contract that is missing this annotation on _some_ of its routes (although I didn't test that) will probably disable those routes entirely